src/main/java/com/google/devtools/build/remote/client/RemoteClient.java

```diff
@@ -407,5 +450,16 @@ public static void main(String[] args) throws Exception {
       return;
     }
 
+    if (optionsParser.getParsedCommand() == "run") {
+      Action.Builder builder = Action.newBuilder();
+      FileInputStream fin = new FileInputStream(runCommand.file);
+      TextFormat.getParser().merge(new InputStreamReader(fin), builder);
+      if (runCommand.path != null) {
+        client.setupDocker(builder.build(), runCommand.path);
+      } else {
+        client.setupDocker(builder.build(), Files.createTempDir().toPath());
+      }
+      return;
```

Comment:
Some style uber-nits:

This function is getting long, and the sequence of:
```
if (optionsParser.getParsedCommand() == "something")  {
  // do it
  return;
}
```
is getting hard to maintain - for example, in some of them you have a return in the end and in some you don't, and IIUC you don't need a return anyway.

So I suggest doing:

```
switch (optionsParser.getParsedCommand()) {
  case "ls":
     doLs(...);
     break;
  case "run":
     doRun(...);
     break;
   ...
}
```

and add a wrapping function for each command. That way, the main is nice and neat and it's obvious what's going on.

Reply:
Thanks for the review! Done.